File: Backend/app/models/notification.py
from tortoise import fields
from tortoise.models import Model
from enum import Enum
import uuid


class NotificationType(str, Enum):
    ORG_INVITE = "org_invite"

class Notification(Model):
    id = fields.UUIDField(pk=True, default=uuid.uuid4)
    user_id = fields.UUIDField(index=True)
    type = fields.CharEnumField(enum_type=NotificationType)
    title = fields.CharField(max_length=512)
    message = fields.TextField()
    metadata = fields.JSONField(null=True) 
    read = fields.BooleanField(default=False)
    created_at = fields.DatetimeField(auto_now_add=True)

    class Meta:
        table = "notifications"

    def __str__(self):
        return f"{self.type} for user {self.user_id}"

File: frontend/src/types/notification.ts
export type NotificationType = "org_invite"

export interface NotificationMetadata {
  org_id?: string
  org_name?: string
  inviter_name?: string
  membership_id?: string
}

export interface Notification {
  id: string
  type: NotificationType | string
  title: string
  message: string
  metadata: NotificationMetadata
  read: boolean
  created_at: string | null
}

export interface NotificationPagination {
  page: number
  page_size: number
  total: number
  total_pages: number
}

export interface NotificationListResponse {
  notifications: Notification[]
  pagination: NotificationPagination
}
